unity-client/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfileController.cs

```diff
@@ -35,8 +36,20 @@ public void LoadProfile(string payload)
         }
 
         var model = JsonUtility.FromJson<UserProfileModel>(payload);
-        ownUserProfile.UpdateData(model);
-        userProfilesCatalog.Add(model.userId, ownUserProfile);
+
+        CatalogController.RequestBaseWearables()
```

Comment:
There's no way of using `IEnumerator` here to avoid the double nesting of `Then`?

Reply:
We could extract both requests into a coroutine function (where we would wait for each request) and start it into `LoadProfile` function, but I don't know if it's worth adding a coroutine execution for this specific case. For me the current code is not adding a hugh complexity that make it not understandable, so in this case I prefer to leave it at that if you don't mind.

Anyway if we consider that it could be a problem in the future, we can change it without problem 👍

Reply:
Thanks Brian!